diff -up openchange-1.0-BORG/libmapi.pc.in.libmapi-conflict openchange-1.0-BORG/libmapi.pc.in
--- openchange-1.0-BORG/libmapi.pc.in.libmapi-conflict	2009-08-05 21:20:11.000000000 +0200
+++ openchange-1.0-BORG/libmapi.pc.in	2012-03-19 11:31:57.845339070 +0100
@@ -8,7 +8,7 @@ datadir=@datadir@
 Name: MAPI
 Description: MAPI Protocol Implementation
 Version: @PACKAGE_VERSION@
-Libs: -L${libdir} -lmapi
+Libs: -L${libdir} -lmapi-openchange
 Libs.private: @LIBS@
 Cflags: -I${includedir}
 Requires: talloc dcerpc ndr tevent
diff -up openchange-1.0-BORG/Makefile.libmapi-conflict openchange-1.0-BORG/Makefile
--- openchange-1.0-BORG/Makefile.libmapi-conflict	2012-03-15 14:50:43.000000000 +0100
+++ openchange-1.0-BORG/Makefile	2012-03-19 11:34:46.894424573 +0100
@@ -159,7 +159,7 @@ LIBMAPI_SO_VERSION = 0
 
 libmapi:	idl					\
 		libmapi/version.h			\
-		libmapi.$(SHLIBEXT).$(PACKAGE_VERSION)	
+		libmapi-openchange.$(SHLIBEXT).$(PACKAGE_VERSION)	
 
 libmapi-install:	libmapi			\
 			libmapi-installpc	\
@@ -192,8 +192,8 @@ endif
 	rm -f *~
 	rm -f */*~
 	rm -f */*/*~
-	rm -f libmapi.$(SHLIBEXT).$(PACKAGE_VERSION) libmapi.$(SHLIBEXT).* \
-		  libmapi.$(SHLIBEXT)
+	rm -f libmapi-openchange.$(SHLIBEXT).$(PACKAGE_VERSION) libmapi-openchange.$(SHLIBEXT).* \
+		  libmapi-openchange.$(SHLIBEXT)
 
 clean:: libmapi-clean
 
@@ -210,10 +210,10 @@ libmapi-installpc:
 libmapi-installlib:
 	@echo "[*] install: libmapi library"
 	$(INSTALL) -d $(DESTDIR)$(libdir)
-	$(INSTALL) -m 0755 libmapi.$(SHLIBEXT).$(PACKAGE_VERSION) $(DESTDIR)$(libdir)
-	ln -sf libmapi.$(SHLIBEXT).$(PACKAGE_VERSION) $(DESTDIR)$(libdir)/libmapi.$(SHLIBEXT)
+	$(INSTALL) -m 0755 libmapi-openchange.$(SHLIBEXT).$(PACKAGE_VERSION) $(DESTDIR)$(libdir)
+	ln -sf libmapi-openchange.$(SHLIBEXT).$(PACKAGE_VERSION) $(DESTDIR)$(libdir)/libmapi-openchange.$(SHLIBEXT)
 ifeq ($(MANUALLY_CREATE_SYMLINKS), yes)
-	ln -sf libmapi.$(SHLIBEXT).$(PACKAGE_VERSION) $(DESTDIR)$(libdir)/libmapi.$(SHLIBEXT).$(LIBMAPI_SO_VERSION)
+	ln -sf libmapi-openchange.$(SHLIBEXT).$(PACKAGE_VERSION) $(DESTDIR)$(libdir)/libmapi-openchange.$(SHLIBEXT).$(LIBMAPI_SO_VERSION)
 endif
 
 libmapi-installheader:
@@ -254,7 +254,7 @@ libmapi-uninstallpc:
 	rm -f $(DESTDIR)$(libdir)/pkgconfig/libmapi.pc
 
 libmapi-uninstalllib:
-	rm -f $(DESTDIR)$(libdir)/libmapi.*
+	rm -f $(DESTDIR)$(libdir)/libmapi-openchange.*
 
 libmapi-uninstallheader:
 	rm -rf $(DESTDIR)$(includedir)/libmapi
@@ -265,7 +265,7 @@ libmapi-uninstallscript:
 	rm -f $(DESTDIR)$(datadir)/setup/profiles/oc_profiles*
 	rm -rf $(DESTDIR)$(datadir)/setup/profiles
 
-libmapi.$(SHLIBEXT).$(PACKAGE_VERSION): 		\
+libmapi-openchange.$(SHLIBEXT).$(PACKAGE_VERSION): 		\
 	libmapi/emsmdb.po				\
 	libmapi/async_emsmdb.po				\
 	libmapi/IABContainer.po				\
@@ -309,10 +309,10 @@ libmapi.$(SHLIBEXT).$(PACKAGE_VERSION):
 	libmapi/socket/interface.po			\
 	libmapi/socket/netif.po				
 	@echo "Linking $@"
-	@$(CC) $(DSOOPT) $(CFLAGS) $(LDFLAGS) -Wl,-soname,libmapi.$(SHLIBEXT).$(LIBMAPI_SO_VERSION) -o $@ $^ $(LIBS)
+	@$(CC) $(DSOOPT) $(CFLAGS) $(LDFLAGS) -Wl,-soname,libmapi-openchange.$(SHLIBEXT).$(LIBMAPI_SO_VERSION) -o $@ $^ $(LIBS)
 
 
-libmapi.$(SHLIBEXT).$(LIBMAPI_SO_VERSION): libmapi.$(SHLIBEXT).$(PACKAGE_VERSION)
+libmapi-openchange.$(SHLIBEXT).$(LIBMAPI_SO_VERSION): libmapi-openchange.$(SHLIBEXT).$(PACKAGE_VERSION)
 	ln -fs $< $@
 
 libmapi/version.h: VERSION
@@ -349,7 +349,7 @@ libmapipp.$(SHLIBEXT).$(PACKAGE_VERSION)
 	libmapi++/src/object.po			\
 	libmapi++/src/profile.po		\
 	libmapi++/src/session.po \
-	libmapi.$(SHLIBEXT).$(LIBMAPI_SO_VERSION)
+	libmapi-openchange.$(SHLIBEXT).$(LIBMAPI_SO_VERSION)
 	@echo "Linking $@"
 	@$(CXX) $(DSOOPT) $(CXXFLAGS) $(LDFLAGS) -Wl,-soname,libmapipp.$(SHLIBEXT).$(LIBMAPIPP_SO_VERSION) -o $@ $^ $(LIBS) 
 
@@ -430,7 +430,7 @@ clean:: libmapixx-tests-clean
 
 bin/libmapixx-test:	libmapi++/tests/test.cpp	\
 		libmapipp.$(SHLIBEXT).$(PACKAGE_VERSION) \
-		libmapi.$(SHLIBEXT).$(PACKAGE_VERSION)
+		libmapi-openchange.$(SHLIBEXT).$(PACKAGE_VERSION)
 	@echo "Linking sample application $@"
 	@$(CXX) $(CXXFLAGS) $(LDFLAGS) -o $@ $^ $(LIBS) 
 
@@ -445,7 +445,7 @@ libmapixx-attach-clean:
 
 bin/libmapixx-attach: libmapi++/tests/attach_test.po	\
 		libmapipp.$(SHLIBEXT).$(PACKAGE_VERSION) \
-		libmapi.$(SHLIBEXT).$(PACKAGE_VERSION)
+		libmapi-openchange.$(SHLIBEXT).$(PACKAGE_VERSION)
 	@echo "Linking sample application $@"
 	@$(CXX) $(CXXFLAGS) -o $@ $^ $(LIBS)
 
@@ -455,7 +455,7 @@ libmapixx-exception: bin/libmapixx-excep
  
 bin/libmapixx-exception: libmapi++/tests/exception_test.cpp \
 		libmapipp.$(SHLIBEXT).$(PACKAGE_VERSION) \
-		libmapi.$(SHLIBEXT).$(PACKAGE_VERSION)
+		libmapi-openchange.$(SHLIBEXT).$(PACKAGE_VERSION)
 	@echo "Linking exception test application $@"
 	@$(CXX) $(CXXFLAGS) -o $@ $^ $(LIBS)
 
@@ -475,7 +475,7 @@ libmapixx-profiletest-clean:
 
 bin/libmapixx-profiletest: libmapi++/tests/profile_test.po	\
 		libmapipp.$(SHLIBEXT).$(PACKAGE_VERSION) \
-		libmapi.$(SHLIBEXT).$(PACKAGE_VERSION)
+		libmapi-openchange.$(SHLIBEXT).$(PACKAGE_VERSION)
 	@echo "Linking profile test application $@"
 	@$(CXX) $(CXXFLAGS) -o $@ $^ $(LIBS)
 
@@ -496,7 +496,7 @@ libmapixx-messages-clean:
 
 libmapi++/examples/foldertree: libmapi++/examples/foldertree.cpp	\
 		libmapipp.$(SHLIBEXT).$(PACKAGE_VERSION) \
-		libmapi.$(SHLIBEXT).$(PACKAGE_VERSION)
+		libmapi-openchange.$(SHLIBEXT).$(PACKAGE_VERSION)
 	@echo "Linking foldertree example application $@"
 	@$(CXX) $(CXXFLAGS) -o $@ $^ $(LIBS)
 
@@ -504,7 +504,7 @@ clean:: libmapixx-foldertree-clean
 
 libmapi++/examples/messages: libmapi++/examples/messages.cpp	\
 		libmapipp.$(SHLIBEXT).$(PACKAGE_VERSION) \
-		libmapi.$(SHLIBEXT).$(PACKAGE_VERSION)
+		libmapi-openchange.$(SHLIBEXT).$(PACKAGE_VERSION)
 	@echo "Linking messages example application $@"
 	@$(CXX) $(CXXFLAGS) -o $@ $^ $(LIBS)
 
@@ -571,7 +571,7 @@ libmapiadmin-uninstallheader:
 libmapiadmin.$(SHLIBEXT).$(PACKAGE_VERSION):	\
 	libmapiadmin/mapiadmin_user.po		\
 	libmapiadmin/mapiadmin.po 		\
-	libmapi.$(SHLIBEXT).$(PACKAGE_VERSION)
+	libmapi-openchange.$(SHLIBEXT).$(PACKAGE_VERSION)
 	@echo "Linking $@"
 	@$(CC) $(DSOOPT) $(LDFLAGS) -Wl,-soname,libmapiadmin.$(SHLIBEXT).$(LIBMAPIADMIN_SO_VERSION) -o $@ $^ $(LIBS) $(LIBMAPIADMIN_LIBS) 
 
@@ -648,7 +648,7 @@ libocpf.$(SHLIBEXT).$(PACKAGE_VERSION):
 	libocpf/ocpf_dump.po			\
 	libocpf/ocpf_api.po			\
 	libocpf/ocpf_write.po			\
-	libmapi.$(SHLIBEXT).$(PACKAGE_VERSION)
+	libmapi-openchange.$(SHLIBEXT).$(PACKAGE_VERSION)
 	@echo "Linking $@"
 	@$(CC) $(DSOOPT) $(LDFLAGS) -Wl,-soname,libocpf.$(SHLIBEXT).$(LIBOCPF_SO_VERSION) -o $@ $^ $(LIBS)
 
@@ -722,7 +722,7 @@ mapiproxy/dcesrv_mapiproxy.$(SHLIBEXT):
 						gen_ndr/ndr_exchange.po				
 
 	@echo "Linking $@"
-	@$(CC) -o $@ $(DSOOPT) $^ -L. $(LDFLAGS) $(LIBS) $(SAMBASERVER_LIBS) -Lmapiproxy mapiproxy/libmapiproxy.$(SHLIBEXT).$(PACKAGE_VERSION) libmapi.$(SHLIBEXT).$(PACKAGE_VERSION)
+	@$(CC) -o $@ $(DSOOPT) $^ -L. $(LDFLAGS) $(LIBS) $(SAMBASERVER_LIBS) -Lmapiproxy mapiproxy/libmapiproxy.$(SHLIBEXT).$(PACKAGE_VERSION) libmapi-openchange.$(SHLIBEXT).$(PACKAGE_VERSION)
 
 mapiproxy/dcesrv_mapiproxy.c: gen_ndr/ndr_exchange_s.c gen_ndr/ndr_exchange.c
 
@@ -766,7 +766,7 @@ mapiproxy/libmapiproxy.$(SHLIBEXT).$(PAC
 							mapiproxy/libmapiproxy/mapi_handles.po			\
 							mapiproxy/libmapiproxy/entryid.po			\
 							mapiproxy/libmapiproxy/modules.po			\
-							libmapi.$(SHLIBEXT).$(PACKAGE_VERSION)
+							libmapi-openchange.$(SHLIBEXT).$(PACKAGE_VERSION)
 	@echo "Linking $@"
 	@$(CC) -o $@ $(DSOOPT) $(LDFLAGS) -Wl,-soname,libmapiproxy.$(SHLIBEXT).$(LIBMAPIPROXY_SO_VERSION) $^ -L. $(LIBS) $(TDB_LIBS)
 
@@ -898,7 +898,7 @@ mapiproxy/libmapistore.$(SHLIBEXT).$(PAC
 							mapiproxy/libmapistore/mapistore_replica_mapping.po		\
 							mapiproxy/libmapistore/mapistore_namedprops.po			\
 							mapiproxy/libmapistore/mapistore_notification.po 		\
-							libmapi.$(SHLIBEXT).$(PACKAGE_VERSION)
+							libmapi-openchange.$(SHLIBEXT).$(PACKAGE_VERSION)
 	@echo "Linking $@"
 	@$(CC) -o $@ $(DSOOPT) $^ -L. $(LDFLAGS) $(LIBS) $(TDB_LIBS) $(DL_LIBS) -Wl,-soname,libmapistore.$(SHLIBEXT).$(LIBMAPISTORE_SO_VERSION)
 
@@ -923,7 +923,7 @@ mapistore_test: bin/mapistore_test
 bin/mapistore_test: 	mapiproxy/libmapistore/tests/mapistore_test.o		\
 			mapiproxy/libmapistore.$(SHLIBEXT).$(PACKAGE_VERSION)
 	@echo "Linking $@"
-	@$(CC) -o $@ $^ $(LDFLAGS) $(LIBS) -lpopt -L. libmapi.$(SHLIBEXT).$(PACKAGE_VERSION)
+	@$(CC) -o $@ $^ $(LDFLAGS) $(LIBS) -lpopt -L. libmapi-openchange.$(SHLIBEXT).$(PACKAGE_VERSION)
 
 mapistore_clean:
 	rm -f mapiproxy/libmapistore/tests/*.o
@@ -1086,7 +1086,7 @@ clean:: openchangeclient-clean
 
 bin/openchangeclient: 	utils/openchangeclient.o			\
 			utils/openchange-tools.o			\
-			libmapi.$(SHLIBEXT).$(PACKAGE_VERSION)		\
+			libmapi-openchange.$(SHLIBEXT).$(PACKAGE_VERSION)		\
 			libocpf.$(SHLIBEXT).$(PACKAGE_VERSION)
 	@echo "Linking $@"
 	@$(CC) $(CFLAGS) -o $@ $^ $(LDFLAGS) $(LIBS) -lpopt
@@ -1115,7 +1115,7 @@ clean:: mapiprofile-clean
 
 bin/mapiprofile: 	utils/mapiprofile.o 			\
 			utils/openchange-tools.o 		\
-			libmapi.$(SHLIBEXT).$(PACKAGE_VERSION)
+			libmapi-openchange.$(SHLIBEXT).$(PACKAGE_VERSION)
 	@echo "Linking $@"
 	@$(CC) $(CFLAGS) -o $@ $^ $(LDFLAGS) $(LIBS) -lpopt
 
@@ -1143,7 +1143,7 @@ clean:: openchangepfadmin-clean
 
 bin/openchangepfadmin:	utils/openchangepfadmin.o			\
 			utils/openchange-tools.o			\
-			libmapi.$(SHLIBEXT).$(PACKAGE_VERSION) 		\
+			libmapi-openchange.$(SHLIBEXT).$(PACKAGE_VERSION) 		\
 			libmapiadmin.$(SHLIBEXT).$(PACKAGE_VERSION)
 	@echo "Linking $@"
 	@$(CC) -o $@ $^ $(LDFLAGS) $(LIBS) $(LIBMAPIADMIN_LIBS) -lpopt			
@@ -1175,7 +1175,7 @@ clean:: exchange2mbox-clean
 
 bin/exchange2mbox:	utils/exchange2mbox.o				\
 			utils/openchange-tools.o			\
-			libmapi.$(SHLIBEXT).$(PACKAGE_VERSION)
+			libmapi-openchange.$(SHLIBEXT).$(PACKAGE_VERSION)
 	@echo "Linking $@"
 	@$(CC) -o $@ $^ $(LIBS) $(LDFLAGS) -lpopt  $(MAGIC_LIBS)
 
@@ -1239,7 +1239,7 @@ bin/exchange2ical:	utils/exchange2ical_t
 			libexchange2ical/ical2exchange.o	\
 			libexchange2ical/ical2exchange_property.o	\
 			utils/openchange-tools.o			\
-			libmapi.$(SHLIBEXT).$(PACKAGE_VERSION)
+			libmapi-openchange.$(SHLIBEXT).$(PACKAGE_VERSION)
 	@echo "Linking $@"
 	@$(CC) $(LDFLAGS) -o $@ $^ $(LIBS) $(ICAL_LIBS) -lpopt
 
@@ -1303,7 +1303,7 @@ bin/mapitest:	utils/mapitest/mapitest.o
 		utils/mapitest/modules/module_lcid.o		\
 		utils/mapitest/modules/module_mapidump.o	\
 		utils/mapitest/modules/module_lzxpress.o	\
-		libmapi.$(SHLIBEXT).$(PACKAGE_VERSION)		
+		libmapi-openchange.$(SHLIBEXT).$(PACKAGE_VERSION)		
 	@echo "Linking $@"
 	@$(CC) -o $@ $^ $(LDFLAGS) $(LIBS) -lpopt $(SUBUNIT_LIBS)
 
@@ -1361,7 +1361,7 @@ clean:: openchangemapidump-clean
 bin/openchangemapidump:	utils/backup/openchangemapidump.o		\
 			utils/backup/openchangebackup.o			\
 			utils/openchange-tools.o			\
-			libmapi.$(SHLIBEXT).$(PACKAGE_VERSION)
+			libmapi-openchange.$(SHLIBEXT).$(PACKAGE_VERSION)
 	@echo "Linking $@"
 	@$(CC) -o $@ $^ $(LDFLAGS) $(LIBS) -lpopt
 
@@ -1412,7 +1412,7 @@ check_fasttransfer-clean::
 clean:: check_fasttransfer-clean
 
 bin/check_fasttransfer:	testprogs/check_fasttransfer.o			\
-			libmapi.$(SHLIBEXT).$(PACKAGE_VERSION)		\
+			libmapi-openchange.$(SHLIBEXT).$(PACKAGE_VERSION)		\
 			mapiproxy/libmapistore.$(SHLIBEXT).$(PACKAGE_VERSION)
 	@echo "Linking $@"
 	@$(CC) -o $@ $^ $(LIBS) $(LDFLAGS) -lpopt
@@ -1439,7 +1439,7 @@ test_asyncnotif-clean::
 clean:: test_asyncnotif-clean
 
 bin/test_asyncnotif:	testprogs/test_asyncnotif.o			\
-			libmapi.$(SHLIBEXT).$(PACKAGE_VERSION)
+			libmapi-openchange.$(SHLIBEXT).$(PACKAGE_VERSION)
 	@echo "Linking $@"
 	@$(CC) -o $@ $^ $(LIBS) $(LDFLAGS) -lpopt
 
@@ -1484,13 +1484,13 @@ pyopenchange: 	$(pythonscriptdir)/opench
 
 $(pythonscriptdir)/openchange/mapi.$(SHLIBEXT):	pyopenchange/pymapi.c				\
 						pyopenchange/pymapi_properties.c		\
-						libmapi.$(SHLIBEXT).$(PACKAGE_VERSION)
+						libmapi-openchange.$(SHLIBEXT).$(PACKAGE_VERSION)
 	@echo "Linking $@"
 	@$(CC) $(CFLAGS) -fno-strict-aliasing $(DSOOPT) $(LDFLAGS) -o $@ $^ $(PYTHON_CFLAGS) $(PYTHON_LIBS) $(LIBS) 
 
 # $(pythonscriptdir)/openchange/ocpf.$(SHLIBEXT):	pyopenchange/pyocpf.c				\
 # 						libocpf.$(SHLIBEXT).$(PACKAGE_VERSION)		\
-# 						libmapi.$(SHLIBEXT).$(PACKAGE_VERSION)
+# 						libmapi-openchange.$(SHLIBEXT).$(PACKAGE_VERSION)
 # 	@echo "Linking $@"
 # 	@$(CC) $(CFLAGS) $(DSOOPT) $(LDFLAGS) -o $@ $^ $(PYTHON_CFLAGS) $(PYTHON_LIBS) $(LIBS) 
 
@@ -1608,7 +1608,7 @@ ctags:
 
 test:: check
 
-check:: libmapi.$(SHLIBEXT).$(LIBMAPI_SO_VERSION)
+check:: libmapi-openchange.$(SHLIBEXT).$(LIBMAPI_SO_VERSION)
 	# FIXME: Set up server
 	./bin/mapitest --mapi-calls 
 
@@ -1681,7 +1681,7 @@ libqtmapi.$(SHLIBEXT).$(PACKAGE_VERSION)
 
 qt/demo/demoapp: qt/demo/demoapp.o 				\
 	qt/demo/main.o 					\
-	libmapi.$(SHLIBEXT).$(PACKAGE_VERSION) 		\
+	libmapi-openchange.$(SHLIBEXT).$(PACKAGE_VERSION) 		\
 	libmapipp.$(SHLIBEXT).$(PACKAGE_VERSION)	\
 	libqtmapi.$(SHLIBEXT).$(PACKAGE_VERSION)
 	@echo "Linking $@"
