diff -up openchange-2.0-QUADRANT/configure.ac.OC_RULE_ADD-fix openchange-2.0-QUADRANT/configure.ac
--- openchange-2.0-QUADRANT/configure.ac.OC_RULE_ADD-fix	2013-02-14 10:55:32.037997680 +0100
+++ openchange-2.0-QUADRANT/configure.ac	2013-02-14 10:58:08.012868678 +0100
@@ -657,11 +657,20 @@ AC_SUBST(OC_MAPISTORE_CLEAN)
 AC_SUBST(OC_MAPISTORE_INSTALL)
 AC_SUBST(OC_MAPISTORE_UNINSTALL)
 
+OC_SERVER=
+OC_SERVER_CLEAN=
+OC_SERVER_INSTALL=
+OC_SERVER_UNINSTALL=
+AC_SUBST(OC_SERVER)
+AC_SUBST(OC_SERVER_CLEAN)
+AC_SUBST(OC_SERVER_INSTALL)
+AC_SUBST(OC_SERVER_UNINSTALL)
+
 dnl ##########################################################################
 dnl mapiproxy server
 dnl ##########################################################################
 if test x$PYTHON != x; then
-	if test "x$SAMBASERVER_LIBS" != x ; then
+	if test "x$SAMDB_LIBS" != x -a "x$SAMBASERVER_LIBS" != x ; then
 		mapiproxy=1
 	fi
 fi
@@ -684,10 +693,16 @@ AC_ARG_ENABLE(pymapi, AC_HELP_STRING([--
 PYCDIR=`$PYTHON -c "import distutils.sysconfig; print distutils.sysconfig.get_python_lib(1, prefix='\\$(prefix)')"`
 AC_SUBST(PYCDIR)
 
-dnl Enable pyopenchange by default
-PYMAPIALL="$PYMAPIALL pyopenchange"
-PYMAPIINSTALL="$PYMAPIALLINSTALL pyopenchange-install"
-PYMAPIUNINSTALL="$PYMAPIUNINSTALL pyopenchange-uninstall"
+PYMAPIALL=
+PYMAPIINSTALL=
+PYMAPIUNINSTALL=
+
+if test x$enable_pymapi = xyes ; then
+	dnl Enable pyopenchange by default
+	PYMAPIALL="$PYMAPIALL pyopenchange"
+	PYMAPIINSTALL="$PYMAPIALLINSTALL pyopenchange-install"
+	PYMAPIUNINSTALL="$PYMAPIUNINSTALL pyopenchange-uninstall"
+fi
 
 AC_SUBST(PYMAPIALL)
 AC_SUBST(PYMAPIINSTALL)
