diff -up openchange-2.0-QUADRANT/script/mkversion.sh.multilib-issue-libmapi-version-h openchange-2.0-QUADRANT/script/mkversion.sh
--- openchange-2.0-QUADRANT/script/mkversion.sh.multilib-issue-libmapi-version-h	2013-01-24 19:55:36.000000000 +0100
+++ openchange-2.0-QUADRANT/script/mkversion.sh	2013-09-19 15:00:43.106758507 +0200
@@ -80,9 +80,14 @@ OPENCHANGE_SYS_KERNEL_NAME=`uname -s`
 OPENCHANGE_SYS_KERNEL_RELEASE=`uname -r`
 OPENCHANGE_SYS_PROCESSOR=`uname -p`
 
-echo "#define OPENCHANGE_SYS_KERNEL_NAME \"${OPENCHANGE_SYS_KERNEL_NAME}\"" >> $TMPFILE
-echo "#define OPENCHANGE_SYS_KERNEL_RELEASE \"${OPENCHANGE_SYS_KERNEL_RELEASE}\"" >> $TMPFILE
-echo "#define OPENCHANGE_SYS_PROCESSOR \"${OPENCHANGE_SYS_PROCESSOR}\"" >> $TMPFILE
+# skip for a multilib issues
+# echo "#define OPENCHANGE_SYS_KERNEL_NAME \"${OPENCHANGE_SYS_KERNEL_NAME}\"" >> $TMPFILE
+# echo "#define OPENCHANGE_SYS_KERNEL_RELEASE \"${OPENCHANGE_SYS_KERNEL_RELEASE}\"" >> $TMPFILE
+# echo "#define OPENCHANGE_SYS_PROCESSOR \"${OPENCHANGE_SYS_PROCESSOR}\"" >> $TMPFILE
+echo "/* 'Unset' to avoid multilib issues */" >> $TMPFILE
+echo "#define OPENCHANGE_SYS_KERNEL_NAME \"Unset\"" >> $TMPFILE
+echo "#define OPENCHANGE_SYS_KERNEL_RELEASE \"Unset\"" >> $TMPFILE
+echo "#define OPENCHANGE_SYS_PROCESSOR \"Unset\"" >> $TMPFILE
 
 mv "$TMPFILE" "$OUTPUT_FILE"
 
