diff -up openchange-1.0-BORG/doc/examples/multiple_notif.c.popt openchange-1.0-BORG/doc/examples/multiple_notif.c
--- openchange-1.0-BORG/doc/examples/multiple_notif.c.popt	2012-06-01 09:15:48.638999377 +0200
+++ openchange-1.0-BORG/doc/examples/multiple_notif.c	2012-06-01 09:15:57.488000012 +0200
@@ -39,7 +39,7 @@
 
 #include <libmapi/libmapi.h>
 #include <pthread.h> 
-#include <samba/popt.h>
+#include <popt.h>
 
 #define	DEFAULT_PROFDB	"%s/.openchange/profiles.ldb"
 
diff -up openchange-1.0-BORG/mapiproxy/libmapistore/tests/mapistore_test.c.popt openchange-1.0-BORG/mapiproxy/libmapistore/tests/mapistore_test.c
--- openchange-1.0-BORG/mapiproxy/libmapistore/tests/mapistore_test.c.popt	2012-06-01 09:14:53.149999984 +0200
+++ openchange-1.0-BORG/mapiproxy/libmapistore/tests/mapistore_test.c	2012-06-01 09:14:57.877000291 +0200
@@ -23,7 +23,7 @@
 #include "mapiproxy/libmapistore/mapistore_errors.h"
 #include <talloc.h>
 #include <core/ntstatus.h>
-#include <samba/popt.h>
+#include <popt.h>
 #include <param.h>
 #include <util/debug.h>
 
diff -up openchange-1.0-BORG/testprogs/check_fasttransfer.c.popt openchange-1.0-BORG/testprogs/check_fasttransfer.c
--- openchange-1.0-BORG/testprogs/check_fasttransfer.c.popt	2012-06-01 09:15:31.326999950 +0200
+++ openchange-1.0-BORG/testprogs/check_fasttransfer.c	2012-06-01 09:15:53.320003138 +0200
@@ -23,7 +23,7 @@
 #include "mapiproxy/libmapistore/mapistore.h"
 #include "mapiproxy/libmapistore/mapistore_errors.h"
 
-#include <samba/popt.h>
+#include <popt.h>
 #include <ldb.h>
 #include <talloc.h>
 #include <inttypes.h>
diff -up openchange-1.0-BORG/testprogs/test_asyncnotif.c.popt openchange-1.0-BORG/testprogs/test_asyncnotif.c
--- openchange-1.0-BORG/testprogs/test_asyncnotif.c.popt	2012-06-01 09:15:07.797999784 +0200
+++ openchange-1.0-BORG/testprogs/test_asyncnotif.c	2012-06-01 09:15:13.762001235 +0200
@@ -21,7 +21,7 @@
 
 #include "libmapi/libmapi.h"
 
-#include <samba/popt.h>
+#include <popt.h>
 #include <talloc.h>
 
 static void popt_openchange_version_callback(poptContext con,
diff -up openchange-1.0-BORG/utils/backup/openchangemapidump.c.popt openchange-1.0-BORG/utils/backup/openchangemapidump.c
--- openchange-1.0-BORG/utils/backup/openchangemapidump.c.popt	2012-06-01 09:14:13.805999904 +0200
+++ openchange-1.0-BORG/utils/backup/openchangemapidump.c	2012-06-01 09:14:34.223001494 +0200
@@ -21,7 +21,7 @@
 */
 
 #include "libmapi/libmapi.h"
-#include <samba/popt.h>
+#include <popt.h>
 #include <param.h>
 
 #include "openchangebackup.h"
diff -up openchange-1.0-BORG/utils/exchange2mbox.c.popt openchange-1.0-BORG/utils/exchange2mbox.c
--- openchange-1.0-BORG/utils/exchange2mbox.c.popt	2012-06-01 09:13:06.190001927 +0200
+++ openchange-1.0-BORG/utils/exchange2mbox.c	2012-06-01 09:13:12.998000103 +0200
@@ -20,7 +20,7 @@
 */
 
 #include "libmapi/libmapi.h"
-#include <samba/popt.h>
+#include <popt.h>
 #include <ldb.h>
 
 #include <sys/types.h>
diff -up openchange-1.0-BORG/utils/mapiprofile.c.popt openchange-1.0-BORG/utils/mapiprofile.c
--- openchange-1.0-BORG/utils/mapiprofile.c.popt	2012-06-01 09:12:37.573000643 +0200
+++ openchange-1.0-BORG/utils/mapiprofile.c	2012-06-01 09:12:47.669000286 +0200
@@ -20,7 +20,7 @@
 */
 
 #include "libmapi/libmapi.h"
-#include <samba/popt.h>
+#include <popt.h>
 #include <param.h>
 #include "openchange-tools.h"
 
diff -up openchange-1.0-BORG/utils/mapitest/mapitest.c.popt openchange-1.0-BORG/utils/mapitest/mapitest.c
--- openchange-1.0-BORG/utils/mapitest/mapitest.c.popt	2012-06-01 09:13:47.148999777 +0200
+++ openchange-1.0-BORG/utils/mapitest/mapitest.c	2012-06-01 09:13:58.079000426 +0200
@@ -22,7 +22,7 @@
 #include "utils/mapitest/mapitest.h"
 #include "utils/openchange-tools.h"
 
-#include <samba/popt.h>
+#include <popt.h>
 #include <param.h>
 
 #include "config.h"
diff -up openchange-1.0-BORG/utils/openchangeclient.c.popt openchange-1.0-BORG/utils/openchangeclient.c
--- openchange-1.0-BORG/utils/openchangeclient.c.popt	2012-06-01 09:13:26.798999464 +0200
+++ openchange-1.0-BORG/utils/openchangeclient.c	2012-06-01 09:13:32.781999154 +0200
@@ -23,7 +23,7 @@
 #include "libmapi/libmapi_private.h"
 #include "libmapi/mapi_nameid.h"
 #include "libocpf/ocpf.h"
-#include <samba/popt.h>
+#include <popt.h>
 #include <param.h>
 
 #include "openchangeclient.h"
diff -up openchange-1.0-BORG/utils/openchangepfadmin.c.popt openchange-1.0-BORG/utils/openchangepfadmin.c
--- openchange-1.0-BORG/utils/openchangepfadmin.c.popt	2011-06-27 23:26:39.000000000 +0200
+++ openchange-1.0-BORG/utils/openchangepfadmin.c	2012-06-01 09:10:03.688999996 +0200
@@ -21,7 +21,7 @@
 
 #include "libmapi/libmapi.h"
 #include "libmapiadmin/libmapiadmin.h"
-#include <samba/popt.h>
+#include <popt.h>
 #include <param.h>
 
 #include "openchangepfadmin.h"
