diff -up openchange-1.0-BORG/libmapi/emsmdb.c.uninit-crash openchange-1.0-BORG/libmapi/emsmdb.c
--- openchange-1.0-BORG/libmapi/emsmdb.c.uninit-crash	2012-03-13 17:58:28.000000000 +0100
+++ openchange-1.0-BORG/libmapi/emsmdb.c	2012-07-10 12:05:26.453775843 +0200
@@ -271,6 +271,10 @@ int emsmdb_disconnect_dtor(void *data)
 	struct emsmdb_context	*emsmdb_ctx;
 
 	emsmdb_ctx = (struct emsmdb_context *)provider->ctx;
+
+	if (!emsmdb_ctx)
+		return MAPI_E_SUCCESS;
+
 	emsmdb_disconnect(emsmdb_ctx);	
 
 	talloc_free(emsmdb_ctx->cache_requests);
@@ -283,7 +287,7 @@ int emsmdb_disconnect_dtor(void *data)
 		talloc_free(emsmdb_ctx->info.szDNPrefix);
 	}
 
-	return 0;
+	return MAPI_E_SUCCESS;
 }
 
 
