diff -up openchange-1.0-BORG/exchange.idl.freebusy openchange-1.0-BORG/exchange.idl
--- openchange-1.0-BORG/exchange.idl.freebusy	2013-10-14 22:43:40.155092822 +0200
+++ openchange-1.0-BORG/exchange.idl	2013-10-14 22:45:54.945083605 +0200
@@ -2920,7 +2920,7 @@ System Attendant Private Interface
 	typedef [flag(NDR_NOALIGN)] struct {
 		uint8		RowNoLongerVisible;
 	  	uint8		HasRowData;
-		[flag(NDR_NOALIGN)] DATA_BLOB row;
+		[flag(NDR_REMAINING|NDR_NOALIGN)] DATA_BLOB row;
 	} FindRow_repl;
 
 	/**************************/
diff -up openchange-1.0-BORG/libmapi/freebusy.c.freebusy openchange-1.0-BORG/libmapi/freebusy.c
--- openchange-1.0-BORG/libmapi/freebusy.c.freebusy	2012-02-20 17:19:01.000000000 +0100
+++ openchange-1.0-BORG/libmapi/freebusy.c	2013-10-14 22:43:15.719094493 +0200
@@ -207,7 +207,7 @@ _PUBLIC_ enum MAPISTATUS GetUserFreeBusy
 	OPENCHANGE_RETVAL_IF(!mid || *mid == MAPI_E_NOT_FOUND, MAPI_E_NOT_FOUND, NULL);
 
 	mapi_object_init(&obj_message);
-	retval = OpenMessage(obj_store, *fid, *mid, &obj_message, 0x0);
+	retval = OpenMessage(&obj_exfreebusy, *fid, *mid, &obj_message, 0x0);
 	OPENCHANGE_RETVAL_IF(retval, retval, NULL);
 
 	/* Step 11. Get FreeBusy properties */
